File: ecopoints/app/src/main/java/at/htl/ecopoints/db/services/TripService.java
package at.htl.ecopoints.db.services;

import at.htl.ecopoints.model.Trip;
import okhttp3.Response;

import java.util.List;
import java.util.UUID;

public class TripService extends Service {
    private final String endPoint = "trip";

    public Response createTrip(Trip trip) throws Exception {
        return super.create(trip, endPoint);
    }

    public Response updateTrip(Trip trip, UUID id) throws Exception {
        return super.update(trip, endPoint, id.toString());
    }

    public Response deleteTrip(UUID id) throws Exception {
        return super.delete(endPoint, id.toString());
    }

    public Trip getTripById(UUID id) throws Exception {
        return super.getById(endPoint, id.toString());
    }

    public List<Trip> getAllTrips() throws Exception {
        return super.getAll(endPoint);
    }
}

File: ecopoints/app/src/main/java/at/htl/ecopoints/backendService/TripService.kt
package at.htl.ecopoints.backendService

import android.util.Log
import at.htl.ecopoints.model.Trip
import okhttp3.Response
import java.util.UUID


class TripService: Service() {
    private final val endPoint: String = "trip"

    fun createTrip(trip: Trip): Response{
        return super.create(trip, endPoint)
    }

    fun updateTrip(trip: Trip, id: UUID): Response{
        return super.update(trip, endPoint, id.toString())
    }

    fun deleteTrip(id: UUID): Response{
        return super.delete(endPoint, id.toString())
    }

    fun getTripById(id: UUID): Trip? {
        val trip = super.getById<Trip>(endPoint, id.toString())
        return trip
    }

    fun getAllTrips(): List<Trip>? {
        val trips = super.getAll<Trip>(endPoint)
        return trips
    }
}